lang/typescript/src/extended-address/generateAddress1.ts

```diff
@@ -0,0 +1,25 @@
+import {concatAddressField} from '../utils/address-fields';
+import {getRegionConfig} from '../utils/regions';
+import {Address} from '../types/address';
+
+/**
+ * Generate a concatenated address1 string based on the region specified by country code
+ * @param address Address containing `countryCode` and any number of other
+ * optional address fields
+ * @returns concatenated address string or null if invalid input was sent
+ */
+export function generateAddress1(address: Address): string | null {
+  const config = getRegionConfig(address.countryCode);
+  const address1CombinedFormat = config?.combined_address_format?.address1;
+  const containsAddress1ExtendedFields =
+    address.streetName !== undefined || address.streetNumber !== undefined;
```

Comment:
Can we generalize this to check that `address` contains at least one of the keys listed in `address1CombinedFormat`? Just so we don't hardcode the assumption that only street names/numbers may appear in address1.

Same recommendation for address2 only containing line2 and neighborhood.